### PR TITLE
9 | deduct cookies for buying upgrades

### PR DESCRIPTION
**Pull request by Gulderbone**

### CODE REVIEW

File: app/src/main/java/com/gulderbone/cookieclicker/activities/ItemShop.kt [hunk 1]
```diff
@@ -17,7 +17,7 @@ class ItemShop : AppCompatActivity() {
 
         upgradeButton = findViewById(R.id.upgrade)
 
-        upgradeButton.setOnClickListener { Game.producerPurchased(Grandma()) }
+        upgradeButton.setOnClickListener { Game.producerPurchased(Grandma(), this) }
```
Comment (Gulderbone) [inline app/src/main/java/com/gulderbone/cookieclicker/activities/ItemShop.kt]: After discussing my codebase now I know that I should not send the entire context to functions.

File: app/src/main/java/com/gulderbone/cookieclicker/cookieproducers/Grandma.kt [hunk 2]
```diff
@@ -1,3 +1,3 @@
 package com.gulderbone.cookieclicker.cookieproducers
 
-class Grandma : CookieProducer("Grandma", 5, 50)
\ No newline at end of file
+class Grandma : CookieProducer("Grandma", 1, 100)
```
Comment (Gulderbone) [inline app/src/main/java/com/gulderbone/cookieclicker/cookieproducers/Grandma.kt:3]: I'm thinking about an enum or something like that to keep prices and cmp of all producers in one place.

**Review by Gulderbone — COMMENTED** [review]
look at the comments later.

File: app/src/main/java/com/gulderbone/cookieclicker/Game.kt [hunk 3]
```diff
@@ -52,4 +58,16 @@ object Game {
             cpm += producer.key.cpm * producer.value
         }
     }
+
+    private fun deductCookiesFromScore(cookieProducer: CookieProducer): Boolean {
+        val amount = if (producers.containsKey(cookieProducer)) producers[cookieProducer] else 0
+        val currentProducerPrice = cookieProducer.calculatePrice(amount!!)
+        if (score < currentProducerPrice) {
```
Comment (Mateusz512) [inline app/src/main/java/com/gulderbone/cookieclicker/Game.kt]: test

File: app/src/main/java/com/gulderbone/cookieclicker/cookieproducers/Grandma.kt (hunk 2 above)
Comment (Gulderbone) [reply app/src/main/java/com/gulderbone/cookieclicker/cookieproducers/Grandma.kt:3]: Not today, I'm too lazy. Added to backlog xd.

**Pull request merged**